src/main/java/org/jenkinsci/plugins/github/webhook/subscriber/DuplicateEventsSubscriber.java

```diff
@@ -0,0 +1,137 @@
+package org.jenkinsci.plugins.github.webhook.subscriber;
+
+import static com.google.common.collect.Sets.immutableEnumSet;
+
+import com.google.common.annotations.VisibleForTesting;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.PeriodicWork;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.github.extension.GHEventsSubscriber;
+import org.jenkinsci.plugins.github.extension.GHSubscriberEvent;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.github.GHEvent;
+
+/**
+ * Tracks duplicate {@link GHEvent} triggering actions in Jenkins.
+ * Events are tracked for 10 minutes, with the last detected duplicate reference retained for up to 24 hours
+ * (see {@link #isDuplicateEventSeen}).
+ * <p>
+ * Duplicates are stored in-memory only, so a controller restart clears all entries as if none existed.
+ * Persistent storage is omitted for simplicity, since webhook misconfigurations would likely cause new duplicates.
+ */
+@Extension
+public final class DuplicateEventsSubscriber extends GHEventsSubscriber {
+
+    private static final Logger LOGGER = Logger.getLogger(DuplicateEventsSubscriber.class.getName());
+    private static final long TTL_MILLIS = TimeUnit.MINUTES.toMillis(10);
+    private static final long TWENTY_FOUR_HOURS_MILLIS = TimeUnit.HOURS.toMillis(24);
+    private static final Map<String, Long> EVENT_TRACKER = new ConcurrentHashMap<>();
+    private static volatile TrackedDuplicateEvent lastDuplicate;
+
+    public record TrackedDuplicateEvent(String eventGuid, long lastUpdated, GHSubscriberEvent ghSubscriberEvent) { }
+
+    /**
+     * This subscriber is not applicable to any item
+     *
+     * @param item ignored
+     * @return always false
+     */
+    @Override
+    protected boolean isApplicable(@Nullable Item item) {
+        return false;
+    }
+
+    /**
+     * Subscribes to events that trigger actions in Jenkins, such as repository scans or builds.
+     * <p>
+     * The {@link GHEvent} enum defines about 63 events, but not all are relevant to Jenkins.
+     * Tracking unnecessary events increases memory usage, and they occur more frequently than those triggering any
+     * work.
+     * <p>
+     * <a href="https://docs.github.com/en/webhooks/webhook-events-and-payloads">
+     *     Documentation reference (also referenced in {@link GHEvent})</a>
+     */
```

Comment:
```suggestion
```
since I think this is just copied from super? Use `@inheritDoc` if you expect someone to be reading this class’ Javadoc _and_ you have something to add specific to this implementation.

Reply:
this javadoc is different from that of super.
https://github.com/jenkinsci/github-plugin/blob/d2e9e60422ae6f72b6ead117b7ea7a5f8a34470d/src/main/java/org/jenkinsci/plugins/github/extension/GHEventsSubscriber.java#L109-L114
Now added the `{@inheritDoc}` prior to extending the javadoc.

ty